player/lua/ytdl_hook.lua

```diff
@@ -999,15 +1085,24 @@ function run_ytdl_hook(url)
                     site = entry["webpage_url"]
                 end
 
+                local playlist_url = nil
+
                 -- links without protocol as returned by --flat-playlist
                 if not site:find("://") then
                     -- youtube extractor provides only IDs,
                     -- others come prefixed with the extractor name and ":"
                     local prefix = site:find(":") and "ytdl://" or
                         "https://youtu.be/"
+                    playlist_url = prefix .. site
                     table.insert(playlist, prefix .. site)
```

Comment:
Shouldn't this line have been removed?

Reply:
Oops, sorry, I was sure I deleted it. Should I submit a PR or someone with push access can commit a fix?

Reply:
Missed that too. Maybe it got lost during a round of force pushing. If that's the only thing, I can delete it real quick.

Reply:
That's the only thing I noticed. The link above doesn't work for me with my config, but it does with --no-config, so it must be something specific to me.

Reply:
Went through the changes again. Everything else seems fine to me (or at least not as trivial). That would be the only fixup I think.

Reply:
Thanks for the eagle eye. 6e428c261e1d005832ec969807d0a5a094862974

Reply:
```
mpv --no-config --script-opts=ytdl_hook-try_ytdl_first=yes "https://gofile.io/d/esXGnQ"
```
This doesn't work.

Reply:
I reverted my changes locally and tried it out, it didn't work before my PR either. If you run `yt-dlp -F https://store11.gofile.io/download/cb1cb178-bb31-40ff-bd38-0352f152e3a6/2021-08-22%2015-49-56%20%5B0zk0uc%5D.mp4` you'll see the same playlist content, which is the cause of the loop. It's a separate issue IMO.

Reply:
It's not obvious to me why that option would make a difference. It must be some ordering thing I guess but the hook has the cookies and executes anyway...

Reply:
Apparently what's going on here is that with `try_ytdl_first=no` it gets the playlist entries with yt-dlp and then plays those entries directly through ffmpeg, but with `try_ytdl_first=yes` it always calls yt-dlp, which doesn't have the appropriate cookies set and thus doesn't work.

Reply:
But it's not about cookies. yt-dlp doesn't any cookies by default. yt-dlp's Gofile extractor always creates a new guest account each time (but it doesn't generally matter). It's more of an ordering problem.

Without the `try_ytdl_first=yes` option:
- mpv fails to load `https://gofile.io/d/esXGnQ`
- ytdl_hook processes the URL, sets `stream-open-filename` to the playlist
- mpv successfully loads the first entry `https://store11.gofile.io/download/...`

With the `try_ytdl_first=yes` option:
- ytdl_hook processes `https://gofile.io/d/esXGnQ`, sets `stream-open-filename` to the playlist
- ytdl_hook processes `https://store11.gofile.io/download/...`, sets `stream-open-filename` to the playlist
- ytdl_hook processes `https://store11.gofile.io/download/...`, sets `stream-open-filename` to the playlist
- ytdl_hook processes `https://store11.gofile.io/download/...`, sets `stream-open-filename` to the playlist
- ...

This PR did nothing to address this problem, yeah, but I think it's unrelated.

Reply:
It certainly has to do with cookies, otherwise yt-dlp wouldn't always extract the playlist and instead get the url to the actual entry after the first playlist pass.

When I hand it `--ytdl-raw-options-append="cookies=cookies"` it stores the cookies in the file `cookies` and then on the next iteration uses those cookies and it works.

Reply:
Though I guess it's yt-dlp's problem, because when provided a `store11.gofile.io` URL it shouldn't return the parent "file group" as a playlist and instead it should return the actual file requested, as a single format? I think?

Reply:
> It certainly has to do with cookies, otherwise yt-dlp wouldn't always extract the playlist and instead get the url to the actual entry after the first playlist pass.

Hm, true, gofile.io does a redirect if you don't have cookies. I dunno, I guess you could put that option into your config file. Still a separate issue though.

Reply:
If we could hand the cookies as an argument I wouldn't consider it to be separate, but since we can't there is nothing we can do here.

This works for per instance cookies files, but it doesn't clean up after itself. Maybe I'll make a script for that.
```
profile-cond=mp.commandv('change-list', 'ytdl-raw-options', 'append', 'cookies=' .. '/tmp/mpv_cookies_' .. pid)
```